nextclade/Snakefile

```diff
@@ -0,0 +1,375 @@
+        label_key= lambda w: config["builds"][w.lineage][w.segment]["clade_systems"][w.clade].get('key', 'clade')
+    shell:
+        """
+        augur clades --tree {input.tree} \
+            --mutations {input.muts} \
+            --clades {input.clades} \
+            --membership-name {params.membership_key} \
+            --label-name {params.label_key} \
+            --output-node-data {output.node_data} \
+            > /dev/null
+        """
+
+rule virus_specific_jsons:
+    input:
+        auspice_config= "config/auspice_config.json",
+        pathogen = "config/pathogen.json",
+        additional_pathogen="dataset_config/{lineage}/{segment}/{reference}/pathogen.json",
+    output:
+        pathogen = "build/{lineage}/{segment}/{reference}/pathogen.json",
+        auspice = "build/{lineage}/{segment}/{reference}/auspice_config.json",
+    params:
+        clades = lambda w: [config["builds"][w.lineage][w.segment]["clade_systems"][clade].get('key', 'default')
+                            for clade in config["builds"][w.lineage][w.segment]["clade_systems"]]
+    shell:
+        """
+        python3 scripts/merge_jsons.py --lineage {wildcards.lineage} --reference {wildcards.reference} \
+            --segment {wildcards.segment} --clades {params.clades} \
+            --pathogen-jsons {input.pathogen} {input.additional_pathogen} \
+            --auspice-config {input.auspice_config} \
+            --output-pathogen {output.pathogen} \
+            --output-auspice {output.auspice}
+        """
+
+
+
+def get_node_data(w):
+    node_data = [
+        rules.refine.output.node_data,
+        "build/{lineage}/{segment}/{reference}/muts.json".format(**w),
+    ]
+
+    for clade in config["builds"][w.lineage][w.segment]["clade_systems"]:
+        node_data.append("build/{lineage}/{segment}/{reference}/".format(**w) + f'clade_{clade}.json')
+
+    return node_data
+
+
+rule export:
+    message:
+        "Exporting data files for auspice"
+    input:
+        tree=rules.refine.output.tree,
+        metadata=rules.parse.output.metadata,
+        node_data = get_node_data,
+        auspice_config= "build/{lineage}/{segment}/{reference}/auspice_config.json"
+    output:
+        auspice_json="build/{lineage}/{segment}/{reference}/auspice.json",
+    params:
+        fields="region strainName country date EPI_ISL",
+        date=datetime.datetime.utcnow().strftime("%Y-%m-%d"),
+    shell:
+        """
+        AUGUR_RECURSION_LIMIT=10000 \
+        augur export v2 \
+            --tree {input.tree} \
+            --metadata {input.metadata} \
+            --node-data {input.node_data}\
+            --auspice-config {input.auspice_config} \
+            --color-by-metadata {params.fields} \
+            --minify-json \
+            --title "Nextclade reference tree for Influenza type:{wildcards.lineage} segment:{wildcards.segment} with root {wildcards.reference} built on {params.date}" \
```

Comment:
I like this pattern of including the date in the title! We should use this in the main workflow, too...